File: desafio-02/jessizanelato/php/NumerosPrimos.php
<?php

final class NumerosPrimos {
    
    public static function encontrarNumerosPrimos ($inicio = 1, $limite) {

        $numerosPrimos = [];

        for($num = $inicio; $num <= $limite; $num++){

            $divisores = self::encontrarDivisores($num);
        
            if(count($divisores) == 1) {
                $numerosPrimos[] = $num;
            }
            
        }

        return $numerosPrimos;
    }

    private static function encontrarDivisores($numero) {
        $divisor = 2;
        $divisores = [];

        while($numero > 1) {
            while ($numero % $divisor == 0) {
                $numero = $numero / $divisor;
                $divisores[] = $divisor;
            }
            $divisor++;
        }

        return $divisores;
    }

}

File: desafio-02/andretaveira90/java/NumerosPrimos.java
public class NumerosPrimos {
    public static void main(String[] args) {
        for (int numero = 2; numero <= 10000; numero++) {
            boolean ehPrimo = true;
            for (int contador = 2; contador < numero; contador++) {
                if (numero % contador == 0) {
                    ehPrimo = false;
                }
            }
            if (ehPrimo) {
                System.out.println(numero);
            }
        }
    }
}

